pkg/ddevapp/config.go

```diff
@@ -1382,6 +1381,26 @@ func WriteImageDockerfile(fullpath string, contents []byte) error {
 	return nil
 }
 
+// IsAllowedProjectLocation returns an error if the project location is not allowed
+func IsAllowedProjectLocation(app *DdevApp) error {
+	homeDir, _ := os.UserHomeDir()
+	if app.AppRoot == filepath.Dir(globalconfig.GetGlobalDdevDir()) || app.AppRoot == homeDir {
+		return fmt.Errorf("'ddev config' is not useful in your home directory (%v)", homeDir)
+	}
+	projectList := globalconfig.GetGlobalProjectList()
+	for _, project := range projectList {
+		if app.AppRoot == project.AppRoot {
+			return nil
+		}
+		// Do not allow 'ddev config' in any parent directory of any project
+		rel, err := filepath.Rel(app.AppRoot, project.AppRoot)
+		if err == nil && !strings.HasPrefix(rel, "..") {
+			return fmt.Errorf("'ddev config' is not allowed in the %s directory because it has a project in the subdirectory %s\nTo use 'ddev config' here, please 'ddev delete' the project in the subdirectory first", app.AppRoot, project.AppRoot)
```

Comment:
`ddev delete` is not really required, and does more than expected. `ddev stop --unlist` ? Or should we just always assume that delete is the best answer?

Reply:
I don't know yet, I will think about it.